src/java/fr/paris/lutece/plugins/workflow/modules/formspdf/service/HtmlToPDFGenerator.java

```diff
@@ -195,23 +283,185 @@ private void writeExportFile( Path directoryFile ) throws IOException
+                String date = parts[0];
+                String time = parts[1];
+                model.put( infoMarker.getMarker(), date + " " + time );
+            }
+            if(infoMarker.getMarker().equals(CONSTANT_FORM_TITLE))
+            {
+                model.put( infoMarker.getMarker(), FormHome.findByPrimaryKey(_formResponse.getFormId()).getTitle());
+            }
+
+        }
+        return model;
+    }
+
+    /**
+     * Gets the response value.
+     *
+     * @param formQuestionResponse
+     *            the form question response
+     * @return List<String> response value
+     */
+    public List<String> getResponseValue( FormQuestionResponse formQuestionResponse )
+    {
+
+        IEntryTypeService entryTypeService ;
+        List<String> listResponseValue = new ArrayList<>( );
+        if(formQuestionResponse != null && formQuestionResponse.getQuestion( ) != null && formQuestionResponse.getQuestion( ).getEntry( ) != null && formQuestionResponse.getEntryResponse() != null)
+        {
+            Entry entry = formQuestionResponse.getQuestion( ).getEntry( );
+            entryTypeService = EntryTypeServiceManager.getEntryTypeService( entry );
+            if ( entryTypeService instanceof AbstractEntryTypeComment )
+            {
+                return listResponseValue;
+            }
+            if ( entryTypeService instanceof EntryTypeTermsOfService )
+            {
+                boolean aggrement = formQuestionResponse.getEntryResponse( ).stream( )
+                        .filter( response -> response.getField( ).getCode( ).equals( EntryTypeTermsOfService.FIELD_AGREEMENT_CODE ) )
+                        .map( Response::getResponseValue ).map( Boolean::valueOf ).findFirst( ).orElse( false );
+
+                if ( aggrement )
+                {
+                    listResponseValue.add( I18nService.getLocalizedString( KEY_LABEL_YES, I18nService.getDefaultLocale( ) ) );
+                }
+                else
+                {
+                    listResponseValue.add( I18nService.getLocalizedString( KEY_LABEL_NO, I18nService.getDefaultLocale( ) ) );
+                }
+                return listResponseValue;
+
+            }
+            for ( Response response : formQuestionResponse.getEntryResponse( ) )
+            {
+                if ((entryTypeService instanceof EntryTypeImage || entryTypeService instanceof EntryTypeCamera))
+                {
+                    PhysicalFile physicalFile = PhysicalFileHome.findByPrimaryKey(Integer.parseInt(response.getFile().getFileKey()));
+                    if (response.getFile() != null)
+                    {
+                        if (physicalFile != null)
+                        {
+                            String encoded = Base64.getEncoder().encodeToString(physicalFile.getValue());
+                            listResponseValue.add("<div style=\"margin-top: 10px; margin-bottom: 10px;\">"
```

Comment:
All this HTML  code should be written in the template, not here.
(it could be a macro)

Reply:
The template is created by the user with Tinymce editor (he is  making the html and copy/pasting the marks).
So I'm not sure what to do, if I want to follow your code review.

Reply:
Ok i created a template for this entry